planning/planning_validator/autoware_planning_validator_collision_checker/src/collision_checker.cpp

```diff
@@ -0,0 +1,523 @@
+// Copyright 2025 Tier IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "autoware/planning_validator_collision_checker/collision_checker.hpp"
+
+#include "autoware/planning_validator_collision_checker/utils.hpp"
+
+#include <autoware/signal_processing/lowpass_filter_1d.hpp>
+#include <autoware_lanelet2_extension/utility/utilities.hpp>
+#include <autoware_utils/geometry/boost_geometry.hpp>
+#include <autoware_utils/geometry/geometry.hpp>
+#include <autoware_utils/ros/parameter.hpp>
+#include <autoware_utils/ros/update_param.hpp>
+#include <autoware_utils/transform/transforms.hpp>
+
+#include <boost/geometry/algorithms/buffer.hpp>
+#include <boost/geometry/algorithms/disjoint.hpp>
+#include <boost/geometry/algorithms/envelope.hpp>
+#include <boost/geometry/algorithms/union.hpp>
+#include <boost/geometry/strategies/cartesian/buffer_point_square.hpp>
+
+#include <lanelet2_core/Forward.h>
+#include <lanelet2_core/LaneletMap.h>
+#include <lanelet2_core/geometry/BoundingBox.h>
+#include <lanelet2_core/geometry/LineString.h>
+#include <lanelet2_core/geometry/Point.h>
+#include <lanelet2_core/geometry/Polygon.h>
+#include <lanelet2_core/primitives/BoundingBox.h>
+#include <lanelet2_routing/RoutingGraph.h>
+
+#ifdef ROS_DISTRO_GALACTIC
+#include <tf2_eigen/tf2_eigen.h>
+#else
+#include <tf2_eigen/tf2_eigen.hpp>
+#endif
+
+#include <algorithm>
+#include <limits>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+namespace autoware::planning_validator
+{
+using autoware_utils::get_or_declare_parameter;
+
+void CollisionChecker::init(
+  rclcpp::Node & node, const std::string & name,
+  const std::shared_ptr<PlanningValidatorContext> & context)
+{
+  module_name_ = name;
+
+  clock_ = node.get_clock();
+  logger_ = node.get_logger();
+  context_ = context;
+
+  setup_parameters(node);
+
+  setup_diag();
+}
+
+void CollisionChecker::setup_parameters(rclcpp::Node & node)
```

Comment:
How about trying out [generate_parameter_library](https://github.com/PickNikRobotics/generate_parameter_library)? It eliminates the need to define parameter structs manually, and also auto-generates code for tasks like updating all parameters. It also supports parameter validation. That said, it's totally fine to handle this in a separate PR.